{
  "checkpoints": [
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00011\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.025,\n    \"tune_init_std\": 0.001\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.001,\n    \"tune_lr\": 0.025\n  },\n  \"experiment_tag\": \"11_tune_init_std=0.001,tune_lr=0.025\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729040.5273538,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00011_11_tune_init_std=0.001,tune_lr=0.025_2022-05-05_05-37-20\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00011_11_tune_init_std=0.001,tune_lr=0.025_2022-05-05_05-37-20/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1621, in get\\n    raise value.as_instanceof_cause()\\nray.exceptions.RayTaskError(ResourceExhaustedError): \\u001b[36mray::MERA.train_buffered()\\u001b[39m (pid=3460, ip=10.128.0.8, repr=<model_tuning.MERA object at 0x7f1b4981a490>)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 178, in train_buffered\\n    result = self.train()\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 237, in train\\n    result = self.step()\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 117, in step\\n    self.model.run_epoch(batch_size, self.iteration, grad_accumulation=config['data']['grad_accumulation'])\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 99, in run_epoch\\n    self.network.update(train_image_batch, train_label_batch, epoch, apply_grads=False)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 113, in update\\n    loss = self.loss(input_batch, label_batch)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 885, in __call__\\n    result = self._call(*args, **kwds)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 956, in _call\\n    return self._concrete_stateful_fn._call_flat(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 1971, in _call_flat\\n    flat_outputs = forward_function.call(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 591, in call\\n    outputs = execute.execute(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/execute.py\\\", line 59, in quick_execute\\n    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\\ntensorflow.python.framework.errors_impl.ResourceExhaustedError:  OOM when allocating tensor with shape[125,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2] and type complex64 on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc\\n\\t [[{{node StatefulPartitionedCall/einsum_26/Einsum_1}}]]\\nHint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info. This isn't available when running in Eager mode.\\n [Op:__forward_loss_6838]\\n\\nFunction call stack:\\nloss\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}",
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00002\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.005,\n    \"tune_init_std\": 0.05\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.05,\n    \"tune_lr\": 0.005\n  },\n  \"experiment_tag\": \"2_tune_init_std=0.05,tune_lr=0.005\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729019.5344963,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00002_2_tune_init_std=0.05,tune_lr=0.005_2022-05-05_05-36-59\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00002_2_tune_init_std=0.05,tune_lr=0.005_2022-05-05_05-36-59/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1621, in get\\n    raise value.as_instanceof_cause()\\nray.exceptions.RayTaskError(InternalError): \\u001b[36mray::MERA.train_buffered()\\u001b[39m (pid=3477, ip=10.128.0.8, repr=<model_tuning.MERA object at 0x7f752008a5e0>)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 178, in train_buffered\\n    result = self.train()\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 237, in train\\n    result = self.step()\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 117, in step\\n    self.model.run_epoch(batch_size, self.iteration, grad_accumulation=config['data']['grad_accumulation'])\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 99, in run_epoch\\n    self.network.update(train_image_batch, train_label_batch, epoch, apply_grads=False)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 113, in update\\n    loss = self.loss(input_batch, label_batch)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 885, in __call__\\n    result = self._call(*args, **kwds)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 956, in _call\\n    return self._concrete_stateful_fn._call_flat(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 1971, in _call_flat\\n    flat_outputs = forward_function.call(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 591, in call\\n    outputs = execute.execute(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/execute.py\\\", line 59, in quick_execute\\n    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\\ntensorflow.python.framework.errors_impl.InternalError:  Attempting to perform BLAS operation using StreamExecutor without BLAS support\\n\\t [[{{node StatefulPartitionedCall/einsum_33/Einsum}}]] [Op:__forward_loss_6838]\\n\\nFunction call stack:\\nloss\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}",
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00001\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.005,\n    \"tune_init_std\": 0.1\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.1,\n    \"tune_lr\": 0.005\n  },\n  \"experiment_tag\": \"1_tune_init_std=0.1,tune_lr=0.005\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729019.529823,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00001_1_tune_init_std=0.1,tune_lr=0.005_2022-05-05_05-36-59\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00001_1_tune_init_std=0.1,tune_lr=0.005_2022-05-05_05-36-59/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1623, in get\\n    raise value\\nray.exceptions.RayActorError: The actor died because of an error raised in its creation task, \\u001b[36mray::MERA.__init__()\\u001b[39m (pid=3478, ip=10.128.0.8)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 103, in __init__\\n    self.setup(copy.deepcopy(self.config))\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 114, in setup\\n    self.model = Tuning_Model(data_path, digits, val_split, deph_p, num_anc, config, tune_config)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 23, in __init__\\n    super().__init__(data_path, digits, val_split, deph_p, num_anc, -1, -1, config)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/model.py\\\", line 115, in __init__\\n    self.network = network.Network(num_pixels, deph_p, num_anc, init_std, lr, config)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 29, in __init__\\n    self.kraus_ops_1_bd = self.construct_dephasing_multiqubit_kraus(self.num_out_qubits)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 192, in construct_dephasing_multiqubit_kraus\\n    m1 = tf.cast(tf.math.sqrt((2 - self.deph_p) / 2), tf.complex64) * tf.eye(2, dtype=tf.complex64)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/util/dispatch.py\\\", line 206, in wrapper\\n    return target(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/ops/math_ops.py\\\", line 5326, in sqrt\\n    return gen_math_ops.sqrt(x, name)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/ops/gen_math_ops.py\\\", line 10301, in sqrt\\n    _ops.raise_from_not_ok_status(e, name)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/framework/ops.py\\\", line 6941, in raise_from_not_ok_status\\n    six.raise_from(core._status_to_exception(e.code, message), None)\\n  File \\\"<string>\\\", line 3, in raise_from\\ntensorflow.python.framework.errors_impl.ResourceExhaustedError: failed to allocate memory [Op:Sqrt]\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}",
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00003\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.005,\n    \"tune_init_std\": 0.01\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.01,\n    \"tune_lr\": 0.005\n  },\n  \"experiment_tag\": \"3_tune_init_std=0.01,tune_lr=0.005\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729019.5766978,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00003_3_tune_init_std=0.01,tune_lr=0.005_2022-05-05_05-36-59\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00003_3_tune_init_std=0.01,tune_lr=0.005_2022-05-05_05-36-59/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1623, in get\\n    raise value\\nray.exceptions.RayActorError: The actor died because of an error raised in its creation task, \\u001b[36mray::MERA.__init__()\\u001b[39m (pid=3481, ip=10.128.0.8)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 103, in __init__\\n    self.setup(copy.deepcopy(self.config))\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 114, in setup\\n    self.model = Tuning_Model(data_path, digits, val_split, deph_p, num_anc, config, tune_config)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 23, in __init__\\n    super().__init__(data_path, digits, val_split, deph_p, num_anc, -1, -1, config)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/model.py\\\", line 115, in __init__\\n    self.network = network.Network(num_pixels, deph_p, num_anc, init_std, lr, config)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 29, in __init__\\n    self.kraus_ops_1_bd = self.construct_dephasing_multiqubit_kraus(self.num_out_qubits)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 192, in construct_dephasing_multiqubit_kraus\\n    m1 = tf.cast(tf.math.sqrt((2 - self.deph_p) / 2), tf.complex64) * tf.eye(2, dtype=tf.complex64)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/util/dispatch.py\\\", line 206, in wrapper\\n    return target(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/ops/math_ops.py\\\", line 5326, in sqrt\\n    return gen_math_ops.sqrt(x, name)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/ops/gen_math_ops.py\\\", line 10301, in sqrt\\n    _ops.raise_from_not_ok_status(e, name)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/framework/ops.py\\\", line 6941, in raise_from_not_ok_status\\n    six.raise_from(core._status_to_exception(e.code, message), None)\\n  File \\\"<string>\\\", line 3, in raise_from\\ntensorflow.python.framework.errors_impl.ResourceExhaustedError: failed to allocate memory [Op:Sqrt]\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}",
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00004\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.005,\n    \"tune_init_std\": 0.005\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.005,\n    \"tune_lr\": 0.005\n  },\n  \"experiment_tag\": \"4_tune_init_std=0.005,tune_lr=0.005\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729019.581934,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00004_4_tune_init_std=0.005,tune_lr=0.005_2022-05-05_05-36-59\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00004_4_tune_init_std=0.005,tune_lr=0.005_2022-05-05_05-36-59/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1621, in get\\n    raise value.as_instanceof_cause()\\nray.exceptions.RayTaskError(InternalError): \\u001b[36mray::MERA.train_buffered()\\u001b[39m (pid=3480, ip=10.128.0.8, repr=<model_tuning.MERA object at 0x7fcbdb2535e0>)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 178, in train_buffered\\n    result = self.train()\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 237, in train\\n    result = self.step()\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 117, in step\\n    self.model.run_epoch(batch_size, self.iteration, grad_accumulation=config['data']['grad_accumulation'])\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 99, in run_epoch\\n    self.network.update(train_image_batch, train_label_batch, epoch, apply_grads=False)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 113, in update\\n    loss = self.loss(input_batch, label_batch)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 885, in __call__\\n    result = self._call(*args, **kwds)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 956, in _call\\n    return self._concrete_stateful_fn._call_flat(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 1971, in _call_flat\\n    flat_outputs = forward_function.call(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 591, in call\\n    outputs = execute.execute(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/execute.py\\\", line 59, in quick_execute\\n    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\\ntensorflow.python.framework.errors_impl.InternalError:  Attempting to perform BLAS operation using StreamExecutor without BLAS support\\n\\t [[{{node StatefulPartitionedCall/einsum_37/Einsum}}]] [Op:__forward_loss_6838]\\n\\nFunction call stack:\\nloss\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}",
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00005\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.005,\n    \"tune_init_std\": 0.001\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.001,\n    \"tune_lr\": 0.005\n  },\n  \"experiment_tag\": \"5_tune_init_std=0.001,tune_lr=0.005\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729019.5866554,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00005_5_tune_init_std=0.001,tune_lr=0.005_2022-05-05_05-36-59\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00005_5_tune_init_std=0.001,tune_lr=0.005_2022-05-05_05-36-59/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1621, in get\\n    raise value.as_instanceof_cause()\\nray.exceptions.RayTaskError(InternalError): \\u001b[36mray::MERA.train_buffered()\\u001b[39m (pid=3484, ip=10.128.0.8, repr=<model_tuning.MERA object at 0x7f43416235e0>)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 178, in train_buffered\\n    result = self.train()\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 237, in train\\n    result = self.step()\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 117, in step\\n    self.model.run_epoch(batch_size, self.iteration, grad_accumulation=config['data']['grad_accumulation'])\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 99, in run_epoch\\n    self.network.update(train_image_batch, train_label_batch, epoch, apply_grads=False)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 113, in update\\n    loss = self.loss(input_batch, label_batch)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 885, in __call__\\n    result = self._call(*args, **kwds)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 956, in _call\\n    return self._concrete_stateful_fn._call_flat(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 1971, in _call_flat\\n    flat_outputs = forward_function.call(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 591, in call\\n    outputs = execute.execute(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/execute.py\\\", line 59, in quick_execute\\n    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\\ntensorflow.python.framework.errors_impl.InternalError:  Attempting to perform BLAS operation using StreamExecutor without BLAS support\\n\\t [[{{node StatefulPartitionedCall/einsum_18/Einsum}}]] [Op:__forward_loss_6838]\\n\\nFunction call stack:\\nloss\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}",
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00007\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.025,\n    \"tune_init_std\": 0.1\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.1,\n    \"tune_lr\": 0.025\n  },\n  \"experiment_tag\": \"7_tune_init_std=0.1,tune_lr=0.025\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729019.595969,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00007_7_tune_init_std=0.1,tune_lr=0.025_2022-05-05_05-36-59\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00007_7_tune_init_std=0.1,tune_lr=0.025_2022-05-05_05-36-59/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1621, in get\\n    raise value.as_instanceof_cause()\\nray.exceptions.RayTaskError(InternalError): \\u001b[36mray::MERA.train_buffered()\\u001b[39m (pid=3479, ip=10.128.0.8, repr=<model_tuning.MERA object at 0x7f90a9acd430>)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 178, in train_buffered\\n    result = self.train()\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 237, in train\\n    result = self.step()\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 117, in step\\n    self.model.run_epoch(batch_size, self.iteration, grad_accumulation=config['data']['grad_accumulation'])\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 99, in run_epoch\\n    self.network.update(train_image_batch, train_label_batch, epoch, apply_grads=False)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 113, in update\\n    loss = self.loss(input_batch, label_batch)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 885, in __call__\\n    result = self._call(*args, **kwds)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 956, in _call\\n    return self._concrete_stateful_fn._call_flat(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 1971, in _call_flat\\n    flat_outputs = forward_function.call(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 591, in call\\n    outputs = execute.execute(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/execute.py\\\", line 59, in quick_execute\\n    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\\ntensorflow.python.framework.errors_impl.InternalError:  Attempting to perform BLAS operation using StreamExecutor without BLAS support\\n\\t [[{{node StatefulPartitionedCall/einsum_40/Einsum}}]] [Op:__forward_loss_6838]\\n\\nFunction call stack:\\nloss\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}",
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00009\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.025,\n    \"tune_init_std\": 0.01\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.01,\n    \"tune_lr\": 0.025\n  },\n  \"experiment_tag\": \"9_tune_init_std=0.01,tune_lr=0.025\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729040.5126717,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00009_9_tune_init_std=0.01,tune_lr=0.025_2022-05-05_05-37-20\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00009_9_tune_init_std=0.01,tune_lr=0.025_2022-05-05_05-37-20/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1621, in get\\n    raise value.as_instanceof_cause()\\nray.exceptions.RayTaskError(ResourceExhaustedError): \\u001b[36mray::MERA.train_buffered()\\u001b[39m (pid=3483, ip=10.128.0.8, repr=<model_tuning.MERA object at 0x7ef6658b0490>)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 178, in train_buffered\\n    result = self.train()\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 237, in train\\n    result = self.step()\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 117, in step\\n    self.model.run_epoch(batch_size, self.iteration, grad_accumulation=config['data']['grad_accumulation'])\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 99, in run_epoch\\n    self.network.update(train_image_batch, train_label_batch, epoch, apply_grads=False)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 113, in update\\n    loss = self.loss(input_batch, label_batch)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 885, in __call__\\n    result = self._call(*args, **kwds)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 956, in _call\\n    return self._concrete_stateful_fn._call_flat(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 1971, in _call_flat\\n    flat_outputs = forward_function.call(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 591, in call\\n    outputs = execute.execute(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/execute.py\\\", line 59, in quick_execute\\n    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\\ntensorflow.python.framework.errors_impl.ResourceExhaustedError:  OOM when allocating tensor with shape[125,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2] and type complex64 on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc\\n\\t [[{{node StatefulPartitionedCall/einsum_26/Einsum_1}}]]\\nHint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info. This isn't available when running in Eager mode.\\n [Op:__forward_loss_6838]\\n\\nFunction call stack:\\nloss\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}",
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00010\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.025,\n    \"tune_init_std\": 0.005\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.005,\n    \"tune_lr\": 0.025\n  },\n  \"experiment_tag\": \"10_tune_init_std=0.005,tune_lr=0.025\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729040.5205204,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00010_10_tune_init_std=0.005,tune_lr=0.025_2022-05-05_05-37-20\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00010_10_tune_init_std=0.005,tune_lr=0.025_2022-05-05_05-37-20/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1621, in get\\n    raise value.as_instanceof_cause()\\nray.exceptions.RayTaskError(ResourceExhaustedError): \\u001b[36mray::MERA.train_buffered()\\u001b[39m (pid=3463, ip=10.128.0.8, repr=<model_tuning.MERA object at 0x7f7954468490>)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 178, in train_buffered\\n    result = self.train()\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 237, in train\\n    result = self.step()\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 117, in step\\n    self.model.run_epoch(batch_size, self.iteration, grad_accumulation=config['data']['grad_accumulation'])\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 99, in run_epoch\\n    self.network.update(train_image_batch, train_label_batch, epoch, apply_grads=False)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 113, in update\\n    loss = self.loss(input_batch, label_batch)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 885, in __call__\\n    result = self._call(*args, **kwds)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 956, in _call\\n    return self._concrete_stateful_fn._call_flat(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 1971, in _call_flat\\n    flat_outputs = forward_function.call(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 591, in call\\n    outputs = execute.execute(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/execute.py\\\", line 59, in quick_execute\\n    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\\ntensorflow.python.framework.errors_impl.ResourceExhaustedError:  OOM when allocating tensor with shape[125,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2] and type complex64 on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc\\n\\t [[{{node StatefulPartitionedCall/einsum_26/Einsum_1}}]]\\nHint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info. This isn't available when running in Eager mode.\\n [Op:__forward_loss_6838]\\n\\nFunction call stack:\\nloss\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}",
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00006\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.025,\n    \"tune_init_std\": 0.5\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.5,\n    \"tune_lr\": 0.025\n  },\n  \"experiment_tag\": \"6_tune_init_std=0.5,tune_lr=0.025\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729019.591405,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00006_6_tune_init_std=0.5,tune_lr=0.025_2022-05-05_05-36-59\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00006_6_tune_init_std=0.5,tune_lr=0.025_2022-05-05_05-36-59/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1621, in get\\n    raise value.as_instanceof_cause()\\nray.exceptions.RayTaskError(InternalError): \\u001b[36mray::MERA.train_buffered()\\u001b[39m (pid=3482, ip=10.128.0.8, repr=<model_tuning.MERA object at 0x7f366a7475e0>)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 178, in train_buffered\\n    result = self.train()\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 237, in train\\n    result = self.step()\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 117, in step\\n    self.model.run_epoch(batch_size, self.iteration, grad_accumulation=config['data']['grad_accumulation'])\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 99, in run_epoch\\n    self.network.update(train_image_batch, train_label_batch, epoch, apply_grads=False)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 113, in update\\n    loss = self.loss(input_batch, label_batch)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 885, in __call__\\n    result = self._call(*args, **kwds)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 956, in _call\\n    return self._concrete_stateful_fn._call_flat(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 1971, in _call_flat\\n    flat_outputs = forward_function.call(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 591, in call\\n    outputs = execute.execute(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/execute.py\\\", line 59, in quick_execute\\n    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\\ntensorflow.python.framework.errors_impl.InternalError:  Attempting to perform BLAS operation using StreamExecutor without BLAS support\\n\\t [[{{node StatefulPartitionedCall/einsum/Einsum}}]] [Op:__forward_loss_6838]\\n\\nFunction call stack:\\nloss\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}",
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00000\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.005,\n    \"tune_init_std\": 0.5\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.5,\n    \"tune_lr\": 0.005\n  },\n  \"experiment_tag\": \"0_tune_init_std=0.5,tune_lr=0.005\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729019.513692,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00000_0_tune_init_std=0.5,tune_lr=0.005_2022-05-05_05-36-59\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00000_0_tune_init_std=0.5,tune_lr=0.005_2022-05-05_05-36-59/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1623, in get\\n    raise value\\nray.exceptions.RayActorError: The actor died because of an error raised in its creation task, \\u001b[36mray::MERA.__init__()\\u001b[39m (pid=3475, ip=10.128.0.8)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 103, in __init__\\n    self.setup(copy.deepcopy(self.config))\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 114, in setup\\n    self.model = Tuning_Model(data_path, digits, val_split, deph_p, num_anc, config, tune_config)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 23, in __init__\\n    super().__init__(data_path, digits, val_split, deph_p, num_anc, -1, -1, config)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/model.py\\\", line 115, in __init__\\n    self.network = network.Network(num_pixels, deph_p, num_anc, init_std, lr, config)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 29, in __init__\\n    self.kraus_ops_1_bd = self.construct_dephasing_multiqubit_kraus(self.num_out_qubits)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 192, in construct_dephasing_multiqubit_kraus\\n    m1 = tf.cast(tf.math.sqrt((2 - self.deph_p) / 2), tf.complex64) * tf.eye(2, dtype=tf.complex64)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/util/dispatch.py\\\", line 206, in wrapper\\n    return target(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/ops/math_ops.py\\\", line 5326, in sqrt\\n    return gen_math_ops.sqrt(x, name)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/ops/gen_math_ops.py\\\", line 10301, in sqrt\\n    _ops.raise_from_not_ok_status(e, name)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/framework/ops.py\\\", line 6941, in raise_from_not_ok_status\\n    six.raise_from(core._status_to_exception(e.code, message), None)\\n  File \\\"<string>\\\", line 3, in raise_from\\ntensorflow.python.framework.errors_impl.ResourceExhaustedError: failed to allocate memory [Op:Sqrt]\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}",
    "{\n  \"trainable_name\": \"MERA\",\n  \"trial_id\": \"61f2e_00008\",\n  \"config\": {\n    \"num_anc\": 0,\n    \"deph_p\": 1,\n    \"tune_lr\": 0.025,\n    \"tune_init_std\": 0.05\n  },\n  \"local_dir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1\",\n  \"evaluated_params\": {\n    \"tune_init_std\": 0.05,\n    \"tune_lr\": 0.025\n  },\n  \"experiment_tag\": \"8_tune_init_std=0.05,tune_lr=0.025\",\n  \"location\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"8005953a000000000000008c0e7261792e74756e652e747269616c948c084c6f636174696f6e9493942981947d94288c08686f73746e616d65944e8c03706964944e75622e\"\n  },\n  \"resources\": {\n    \"cpu\": 12,\n    \"gpu\": 1,\n    \"memory\": 0,\n    \"object_store_memory\": 0,\n    \"extra_cpu\": 0,\n    \"extra_gpu\": 0,\n    \"extra_memory\": 0,\n    \"extra_object_store_memory\": 0,\n    \"custom_resources\": {},\n    \"extra_custom_resources\": {}\n  },\n  \"placement_group_factory\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"800595ce000000000000008c1f7261792e74756e652e7574696c732e706c6163656d656e745f67726f757073948c15506c6163656d656e7447726f7570466163746f72799493942981947d94288c085f62756e646c6573945d947d94288c03435055944740280000000000008c0347505594473ff00000000000008c066d656d6f7279944700000000000000008c136f626a6563745f73746f72655f6d656d6f72799447000000000000000075618c095f7374726174656779948c045041434b948c055f6172677394298c075f6b7761726773947d9475622e\"\n  },\n  \"stopping_criterion\": {},\n  \"log_to_file\": [\n    \"stdout\",\n    \"stderr\"\n  ],\n  \"max_failures\": 0,\n  \"last_result\": {},\n  \"last_update_time\": -Infinity,\n  \"metric_analysis\": {},\n  \"n_steps\": [\n    5,\n    10\n  ],\n  \"metric_n_steps\": {},\n  \"export_formats\": [],\n  \"status\": \"ERROR\",\n  \"start_time\": 1651729040.5040448,\n  \"logdir\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00008_8_tune_init_std=0.05,tune_lr=0.025_2022-05-05_05-36-59\",\n  \"runner\": null,\n  \"last_debug\": 0,\n  \"error_file\": \"/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/MERA_61f2e_00008_8_tune_init_std=0.05,tune_lr=0.025_2022-05-05_05-36-59/error.txt\",\n  \"error_msg\": \"Traceback (most recent call last):\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trial_runner.py\\\", line 739, in _process_trial\\n    results = self.trial_executor.fetch_result(trial)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/ray_trial_executor.py\\\", line 746, in fetch_result\\n    result = ray.get(trial_future[0], timeout=DEFAULT_GET_TIMEOUT)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/_private/client_mode_hook.py\\\", line 82, in wrapper\\n    return func(*args, **kwargs)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/worker.py\\\", line 1621, in get\\n    raise value.as_instanceof_cause()\\nray.exceptions.RayTaskError(ResourceExhaustedError): \\u001b[36mray::MERA.train_buffered()\\u001b[39m (pid=3476, ip=10.128.0.8, repr=<model_tuning.MERA object at 0x7f0f65089490>)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 178, in train_buffered\\n    result = self.train()\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/ray/tune/trainable.py\\\", line 237, in train\\n    result = self.step()\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 117, in step\\n    self.model.run_epoch(batch_size, self.iteration, grad_accumulation=config['data']['grad_accumulation'])\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/non_distributed_tunning/model_tuning.py\\\", line 99, in run_epoch\\n    self.network.update(train_image_batch, train_label_batch, epoch, apply_grads=False)\\n  File \\\"/home/qc_whaley/dephased_ttn_project/mera/network.py\\\", line 113, in update\\n    loss = self.loss(input_batch, label_batch)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 885, in __call__\\n    result = self._call(*args, **kwds)\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/def_function.py\\\", line 956, in _call\\n    return self._concrete_stateful_fn._call_flat(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 1971, in _call_flat\\n    flat_outputs = forward_function.call(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/function.py\\\", line 591, in call\\n    outputs = execute.execute(\\n  File \\\"/home/qc_whaley/anaconda3/envs/tf2.6/lib/python3.9/site-packages/tensorflow/python/eager/execute.py\\\", line 59, in quick_execute\\n    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,\\ntensorflow.python.framework.errors_impl.ResourceExhaustedError:  OOM when allocating tensor with shape[125,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2] and type complex64 on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc\\n\\t [[{{node StatefulPartitionedCall/einsum_26/Einsum_1}}]]\\nHint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info. This isn't available when running in Eager mode.\\n [Op:__forward_loss_6838]\\n\\nFunction call stack:\\nloss\\n\",\n  \"trial_name_creator\": null,\n  \"custom_trial_name\": null,\n  \"custom_dirname\": null,\n  \"saving_to\": null,\n  \"remote_checkpoint_dir_prefix\": null,\n  \"sync_to_cloud\": null,\n  \"checkpoint_freq\": 0,\n  \"checkpoint_at_end\": false,\n  \"keep_checkpoints_num\": null,\n  \"checkpoint_score_attr\": \"training_iteration\",\n  \"sync_on_checkpoint\": true,\n  \"checkpoint_manager\": {\n    \"_type\": \"CLOUDPICKLE_FALLBACK\",\n    \"value\": \"80059566010000000000008c1b7261792e74756e652e636865636b706f696e745f6d616e61676572948c11436865636b706f696e744d616e616765729493942981947d94288c146b6565705f636865636b706f696e74735f6e756d94477ff00000000000008c165f636865636b706f696e745f73636f72655f6465736394898c165f636865636b706f696e745f73636f72655f61747472948c12747261696e696e675f697465726174696f6e948c1c6e65776573745f70657273697374656e745f636865636b706f696e749468008c0a436865636b706f696e749493942981947d94288c0773746f72616765948c0a70657273697374656e74948c0576616c7565944e8c06726573756c74947d9475628c195f6e65776573745f6d656d6f72795f636865636b706f696e7494680b2981947d9428680e8c066d656d6f72799468104e68117d9475628c115f626573745f636865636b706f696e7473945d948c0b5f6d656d62657273686970948f9475622e\"\n  },\n  \"restore_path\": null,\n  \"restoring_from\": null,\n  \"num_failures\": 1,\n  \"has_new_resources\": false,\n  \"results\": \"80054e2e\",\n  \"best_result\": \"80054e2e\",\n  \"param_config\": \"80054e2e\",\n  \"extra_arg\": \"80054e2e\",\n  \"_state_json\": null,\n  \"_state_valid\": false\n}"
  ],
  "runner_data": {
    "_max_pending_trials": 105,
    "_metric": "test_accuracy",
    "_total_time": 0,
    "_iteration": 122,
    "_has_errored": true,
    "_fail_fast": false,
    "_server_port": null,
    "_cached_trial_decisions": {},
    "_queued_trial_decisions": {},
    "_updated_queue": true,
    "_should_stop_experiment": false,
    "_local_checkpoint_dir": "/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1",
    "_remote_checkpoint_dir": null,
    "_stopper": {
      "_type": "CLOUDPICKLE_FALLBACK",
      "value": "80059527000000000000008c107261792e74756e652e73746f70706572948c0b4e6f6f7053746f707065729493942981942e"
    },
    "_resumed": false,
    "_start_time": 1651729019.3012152,
    "_last_checkpoint_time": -Infinity,
    "_session_str": "2022-05-05_05-36-59",
    "checkpoint_file": "/home/qc_whaley/dephased_ttn_project/mera/ray_results/anc0_deph1/experiment_state-2022-05-05_05-36-59.json",
    "_checkpoint_period": "auto",
    "launch_web_server": false
  },
  "stats": {
    "start_time": 1651729019.3012152,
    "timestamp": 1651729073.8725202
  }
}